src/cmap/connection.ts

```diff
@@ -355,7 +328,11 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     }
 
     this.socket.destroy();
-    this.controller.abort(error);
+    if (error) {
+      this.error = error;
+      this.dataEvents?.throw(error).then(undefined, () => null); // squash unhandled rejection
+    }
```

Comment:
what happens if we close a connection without an error when an operation is in progress?  Right now, I think we would do nothing.

Should we instead always set an error (creating one if one doesn't exist?)?
```typescript
this.error = error ?? new MongoNetworkError("connection shutdown");
this.dataEvents?.throw(this.error).then(undefined, () => null); // squash unhandled rejection
```

Reply:
Since we even make an error for onClose I'm not sure error can really ever be optional 🤔